### PR TITLE
ByteToMessageDecoder: Default implementation for decodeLast should be provided 

### PR DESCRIPTION
**Pull request by shekhar-rajak**
### Motivation:

Corrected the `decodeLast` function  in EOF framed protocols.

### Modifications:

- Removed the `decodeLast` function from `NIOChatServer/main.swift`
- Updated the `decodeLast` function  in `NIOWebSocket/WebSocketFrameDecoder.swift`

### Result:

Fixes: https://github.com/apple/swift-nio/issues/1416

### CODE REVIEW

File: Sources/NIOWebSocket/WebSocketFrameDecoder.swift [hunk 1]
```diff
@@ -270,6 +270,7 @@ public final class WebSocketFrameDecoder: ByteToMessageDecoder {
     }
 
     public func decodeLast(context: ChannelHandlerContext, buffer: inout ByteBuffer, seenEOF: Bool) throws -> DecodingState {
```
Comment (shekhar-rajak) [inline Sources/NIOWebSocket/WebSocketFrameDecoder.swift]: I think, we can remove this method definition which is same as the default implementation. Am I right ?

**Comment by shekhar-rajak** [comment]
Should we also update [`NIOSingleStepByteToMessageDecoder.decodeLast`](https://github.com/apple/swift-nio/blob/master/Sources/NIO/SingleStepByteToMessageDecoder.swift#L66) function ?

File: Sources/NIOWebSocket/WebSocketFrameDecoder.swift (hunk 1 above)
Comment (weissi) [reply Sources/NIOWebSocket/WebSocketFrameDecoder.swift]: Yes, agreed

**Comment by shekhar-rajak** [comment]
> Let’s remove the websocket frame decoder decodeLast.

Thanks for the review, I have done the changes.

**Pull request merged**